### ISSUE TITLE
Create comparison module for generating metrics of spectral similarity etc

### ISSUE THREAD

**Issue by raar1** [aiproteomics/aiproteomics#22]
We need a `comparison` module in the package, containing one or more tools that assess:

1. How well does an RT model fit with a given validation set
2. How well does a fragmentation model fit with a given validation set

It seems likely we could use [scikit-learn](https://scikit-learn.org/stable/index.html) for some of the metric calculations and possibly to make nice plots too.

For metrics of similarity between spectra, we can apparently make use of [matchms](https://github.com/matchms/matchms) to calculate a similarity score.
